src/abi/json-abi-format.md

```diff
@@ -26,51 +32,54 @@ we define the following expressions:
 
 The ABI of a contract is represented as a JSON object containing the following properties:
 
-- `"types`": an array describing all the _type declarations_ used (or transitively used) in the ABI. Each _type declaration_ is a JSON object that contains the following properties:
-  - `"typeId"`: a unique integer ID.
+- `"specVersion"`: a string representing the version pointing to this document versioning. When `specVersion` is incremented, it will usually incur an increment in `abiVersion` too.
+- `"abiVersion"`: a string representing the version of the implementation of this document `abiVersion` may be incremented while `specVersion` is not.
```

Comment:
In which situation should one care for `abiVersion`?

Reply:
We could have breaking changes in the spec without breaking chances in the ABI and vice versa.

Also, a new contract may want to deliberately use an old ABI version for compatibility reasons.

I really think these should be different.

Reply:
So, if the spec is at version `3` but users deliberately chose to use version `1`:

```json
{
  "specVersion": "3",
  "abiVersion": "1"
}
```

Reply:
@nedsalk @hal3e Did you understand the distinction between `specVersion` and `abiVersion`?

I suspect I might still be misinterpreting it.

Reply:
@IGI-111 can you please give us an example of a breaking change in the spec that does not break the abi and vice versa ?

Reply:
> Something that would bump the abiVersion would be for instance changing how intrinsic types are encoded

In this case, what's the difference between `abiVersion` and `encoding`?

Reply:
If we already have an encoding field, we don't need an abiVersion one I think.

Reply:
@esdrubal @IGI-111 It is a NIT, but should we use the same suffix for both?
- `specVersion`
- `encodingVersion`

Renaming `encoding` -> `encodingVersion` is pretty easy for us.

Reply:
Thank you.